Fixes First Ctrl+R and Ctrl+F not showing long name variables correctly

piyushdagar opened the pull request:
Fixes First Ctrl+R and Ctrl+F not showing long name variables correctly #26982

willnationsdev committed:
Fix inherited icons for script classes

danielspaniol committed:
Require `return` in all match branches

Before the parser only checked if the catch-all branch has a return in
order to determine if the entire match block has a return.

This code block was assumed to always return.

    match value:
        "test":
            print("test")
        _:
            return

Now as soon as one of the branches has no return, the entire match block
is marked to not have a return.

RandomShaper committed:
Create class for shared memory blocks [wip]

RandomShaper committed:
Create live view dock [wip]

qarmin committed:
Do not lose focus when moving through scene tree

RandomShaper committed:
Drop mouseover when mouse not in the window

Not that this is critical, but it makes behavior of GUI and 2D/3D picking consistent among them.

This also contributes to dropping GUI focus when the mouse leaves the window (in addition to when it loses focus).

bruvzg committed:
[macOS] Fix hiDPI scaling support in `OS.get_real_window_size` and `OS.set_window.size` functions.

KoBeWi committed:
Changed 'Custom Node' to 'Other Node' to be less confusing

piyushdagar committed:
fixes #26982

fixes First Ctrl+R and Ctrl+F not showing long name variables correctly #26982

YeldhamDev committed:
Make size cache dirty when removing tiles in 'TileMap'

fire committed:
Fix scaling issue for texture button focus texture.

Faless committed:
Enabled logging of failed image download.

bojidar-bg committed:
Fix GDScriptCompiler bugging out with identically-named inner class

Fixes #27111

akien-mga commented on scene/gui/line_edit.cpp:
```diff
@@ -1180,7 +1180,8 @@ void LineEdit::set_cursor_position(int p_pos) {
 				if (accum_width > window_width)
 					break;
-
+				if(accum_width < window_width)
```
Style issue: https://travis-ci.org/godotengine/godot/jobs/505833950

guilhermefelipecgs committed:
StateMachine: Fix sync mode

Chaosus committed:
Properly setup seed in RNG

Paulb23 committed:
Fix script editor attempting to set current line to -1 on script open

goodyttoor committed:
Fix order of transformation in GLTF import 

From the gltf 2 spec, the order is R  * S. Previously we did S * R, which broke some mesh with non-uniform scale.

Fix #23356, Fix #14725

karroffel committed:
fix GDNative binding generation for object types

piyushdagar commented:
@akien-mga

piyushdagar commented:
@akien-mga

piyushdagar committed:
Fixed CollisionObject signals do not trigger on Area

Kanabenki committed:
Add missing check for 2d physics bounce and friction setters
The override check was already present for 3d physics but missing for 2d

neikeq committed:
C#: Bindings generator now translates BBCode docs to XML comments

Kanabenki committed:
Keep binds for connected signals when changing node type

qarmin committed:
Fix inspector update after changing anchor via buttons

kiidmale committed:
Ensure non-emitting particles not processed on entering tree

Also removed non-active CPUParticles(3d) from render list
Fixes #27066

shartte committed:
Fix parsing of generic type declarations in C# source files.

shartte committed:
Add support for new MSBuild directory naming introduced in VS 2019.

RandomShaper committed:
Revert accidental commits

This reverts commit fb37284c027b494ed3ec21124001fcb729f42cc4.
This reverts commit 4db0f51b9aa76cfc7649787fe1970af606ce8dab.

aqnuep committed:
Fixed handling of depth texture so it's resolved and bound when needed

- Cleaned up and improved the code determining when we need to use a depth
  prepass (previously it wasn't executed in certain cases even if it was
  needed)
- Added code to prepare and bind the depth texture even when no depth prepass
  or MRTs (more precisely effect buffers) are used

Fixes #25870, #25535, and #25387.

Commit:
[GDNative] remove spam at editor unfocus when using NativeScript

Commit:
[GDNative] fix NativeScript leak in editor

KoBeWi committed:
Better documentation for some move_and_slide parameters

SpechtMagnus committed:
Fixed unfinished max_lines handling for crash avoidance.

zombietype committed:
Prevents crash on loading unrecognized resources.

Editor crashes on trying to load resources with no loaders.
Simple check on the resource loader prevents using a null
resource loader.

Fixes: #27385

cbscribe committed:
[DOC] Physics[2D]DirectBodyState updates

LeonardMeagher2 committed:
Apply proper offset to CSGMesh material and smooth group

Faless committed:
Add object encoding param to serialization methods

Network peers get_var/put_var
File get_var/store_var
GDScript/Mono/VisualScript bytes2var/var2bytes
Add MultiplayerAPI.allow_object_decoding member which deprecates PacketPeer.allow_object_decoding.

Break ABI compatibaility (API compatibility for GDNative).

reduz committed:
Add a "Request Docs" button to code editor.

Commit:
Corrects small typo in atan2() documentation.

Faless committed:
Use same boolean for objects encode and decode.

In a very unintuitive move encode needed false to encode an object,
decode needed true to decode it.
They now need the same value: `true`.

Faless committed:
Multiplayer API now respects allow_object_decoding

Add doc about allow_object_decoding in PacketPeer

akien-mga committed:
i18n: Sync translations with Weblate

Also sync translation template.

neikeq committed:
C#: Add DynamicGodotObject class

Expands to Object.call, Object.set and Object.get for accessing members. This means it can also access members from scripts written in other languages, like GDScript.

clayjohn committed:
keep DEPTH from causing compile error in GLES2

Paulb23 committed:
Fixed not deselecting when clearing lineedit.

guilhermefelipecgs committed:
Fix wrong blend of animation tree

Interpolation cannot use zero values, must use the values from the
animation to be blended.

akien-mga committed:
doc: Fix formatting bug from #27429

akien-mga committed:
doc: Bump version to 3.2

reduz committed:
Some improvements to is_equal_approx, restored Quat operator.

akien-mga commented:
Could you amend the commit log to state what it does in the first line (commit title)? That would be more explicit than just the issue number when reading the `git log`.

Just inverting the two lines should do it:
```
Fix LineEdit not showing long strings correctly

Fixes #26982
```
Or anything that better describes the actual fix, and not just the original issue (Ctrl+F or R in the search bar are not really related to the actual fix. The commit should describe what the fix does, not one of the symptoms of the bug it fixed).
